compiler/rustc_parse/src/parser/cfg_select.rs

```diff
@@ -23,19 +24,26 @@ pub struct CfgSelectBranches {
     pub unreachable: Vec<(CfgSelectPredicate, TokenStream, Span)>,
 }
 
-/// Parses a `TokenTree` that must be of the form `{ /* ... */ }`, and returns a `TokenStream` where
-/// the surrounding braces are stripped.
+/// Parses a `TokenTree` consisting either of `{ /* ... */ }` (and strip the braces) or an
+/// expression followed by a comma (and strip the comma).
```

Comment:
Question: do you know why `cfg_select!` required braces previously? Did this have any grammar ambiguity problems?

Reply:
Interesting. This would probably require https://github.com/rust-lang/rust/pull/144323 to change again 🤔

Reply:
Changing it is fine, but I don't really see a way here to decide whether to format with the comma or as a block? For `match` that is based on whether there is a statement (i.e. a `;`) in the rhs, here I don't see how we can decide when the `,` variant is desired, unless the formatter should always use a `,` in expression position?

Alternatively we just stick with the user's formatting choice, in which case that information needs to be stored.